src/store/actions/user.actions.ts

```diff
@@ -56,17 +66,31 @@ export const addIdentifier = (args: IAddIdentifierArgs): ThunkAction<Promise<voi
 };
 
 export const setActiveUser = (userId: string): ThunkAction<Promise<void>, RootState, unknown, Action> => {
-  return async (dispatch: ThunkDispatch<RootState, unknown, Action>) => {
+  return async (dispatch: ThunkDispatch<RootState, unknown, Action>, getState: CombinedState<any>) => {
     dispatch({type: USERS_LOADING});
-    getUsersFromStorage()
-      .then((users: Map<string, IUser>) => {
+    await getUsersFromStorage()
+      .then(async (users: Map<string, IUser>) => {
         const user = users.get(userId);
         if (user) {
           dispatch({type: SET_ACTIVE_USER_SUCCESS, payload: user});
+          let userState: IUserState = getState().user;
+          while (!userState.activeUser) {
+            await new Promise((resolve) => setTimeout(resolve, 50));
```

Comment:
Probably wise to add an additional guard here. If it cannot get the user after let's say 10 secs, it will never happen and now we would be stuck in this loop. Adding a guard and then a final check whether the userState is not undefined would be wise.

Similar issue for the contact state below

Reply:
Added a max timeout for the infinite loop. Even though I am pretty sure the user can never be stuck on this. Setting the active user is not the problem here and that will trigger the navigation to navigate to the home screen. the worst that could happen is that there are no contacts and vc's being displayed. Besides that this action is being dispatched, so it will not stop the app.